ground/src/main/java/com/google/android/ground/repository/SubmissionRepository.kt

```diff
@@ -137,21 +137,24 @@ constructor(
    * been marked as [SyncStatus.COMPLETED], including pending, in progress, and failed mutations. A
    * new list is emitted on each subsequent change.
    */
-  fun getIncompleteSubmissionMutationsOnceAndStream(
+  suspend fun getIncompleteSubmissionMutationsOnceAndStream(
     surveyId: String,
     locationOfInterestId: String
-  ): Flowable<List<SubmissionMutation>> =
-    rxMaybe { localSurveyStore.getSurveyByIdSuspend(surveyId) }
-      .toFlowable()
-      .flatMap {
-        localSubmissionStore.getSubmissionMutationsByLocationOfInterestIdOnceAndStream(
-          it,
-          locationOfInterestId,
-          MutationEntitySyncStatus.PENDING,
-          MutationEntitySyncStatus.IN_PROGRESS,
-          MutationEntitySyncStatus.FAILED
-        )
-      }
+  ): Flow<List<SubmissionMutation>> {
+    val survey = localSurveyStore.getSurveyByIdSuspend(surveyId)
+
+    return if (survey != null) {
+      localSubmissionStore.getSubmissionMutationsByLocationOfInterestIdOnceAndStream(
+        survey,
+        locationOfInterestId,
+        MutationEntitySyncStatus.PENDING,
+        MutationEntitySyncStatus.IN_PROGRESS,
+        MutationEntitySyncStatus.FAILED
+      )
+    } else {
+      flow {}
+    }
```

Comment:
Instead of branching, can we return early if survey is null?

Reply:
Yes, we should! Done